karapace/schema_registry_apis.py

```diff
@@ -1111,7 +1111,7 @@ async def subjects_schema_post(
                 parsed_typed_schema = ParsedTypedSchema.parse(
                     schema_version.schema.schema_type,
                     schema_version.schema.schema_str,
-                    references=references,
+                    references=schema_version.references,
```

Comment:
This is a fix to the second bug. The previous code used the references given as argument where this object represent the known schema version.